wrpc/peer.go

```diff
@@ -62,8 +64,13 @@ func (p *Peer) AddConn(conn *Conn) {
 		if err != nil {
 			p.ErrLogger(fmt.Errorf("read thread: %w", err))
 		}
-		// TODO(hbagdi): handle error
-		_ = p.conn.Close()
+		err2 := p.conn.Close()
+		if err2 != nil {
+			p.ErrLogger(fmt.Errorf("closing wrpc connection: %w", err2))
+		}
+		if err == nil && err2 == nil {
+			p.ErrLogger(nil)
+		}
```

Comment:
Why even call ErrLogger in this case?

Reply:
to let the Node know the connection was closed, even without error.   in kong/koko#288 the ErrLogger callback is used to remove the node from the manager.

Reply:
Can we have a CloseCallback instead?
Monitoring a logger to detect such a condition is smelly. Really smell.
WDYT?

Reply:
added back.